### PR TITLE
feat(lsp): Explicit type annotation code action

### PR DESCRIPTION
**Pull request by alex-snezhko**
Screenshot from 2024-07-25 21-38-55

Screenshot from 2024-07-25 21-39-01

Screenshot from 2024-07-25 21-39-09

### CODE REVIEW

**Review by marcusroberts — APPROVED** [review]
Great work Alex!  A really useful feature to add too, thank you!

File: compiler/src/language_server/code_action.re [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  type t = option(list(code_action));
+};
+
+let send_no_result = (~id: Protocol.message_id) => {
+  Protocol.response(~id, `Null);
+};
+
+let explicit_type_annotation = (range, uri, version, type_str) => {
+  ResponseResult.{
+    title: "Explicitly annotate type",
```
Comment (ospencer) [inline compiler/src/language_server/code_action.re:43]: What do you think of just `Annotate type`? If there's no type annotation, then adding one would just be an annotation.

**Review by ospencer — COMMENTED** [review]
Two questions for you: what's with the version additions, and what happens if you use this code action on something that already has an annotation?

File: compiler/src/language_server/code_action.re (hunk 1 above)
Comment (alex-snezhko) [reply compiler/src/language_server/code_action.re]: Changed to this

**Comment by alex-snezhko** [comment]
@ospencer I got rid of the version stuff and it doesn't look like it made a difference actually (at least in VSCode's LSP client); I saw in the LSP documentation that the version was one of the response properties so that's why I implemented it initially.

The code action dialog show no longer show if it's tried on something that's already annotated; I realized that there was actually a flaw in my implementation which sometimes showed the code action dialog on something that was already annotated, but I patched it.

**Review by ospencer — APPROVED** [review]
Looks great!

**Pull request merged**